presets/ragengine/main.py

```diff
@@ -71,10 +71,25 @@ async def query_index(request: QueryRequest):
             status_code=500, detail=f"An unexpected error occurred: {str(e)}"
         )
 
-@app.get("/indexed-documents", response_model=ListDocumentsResponse)
-async def list_all_indexed_documents():
+@app.get("/indexes", response_model=List[str])
+def list_indexes():
     try:
-        documents = rag_ops.list_all_indexed_documents()
+        return rag_ops.list_indexes()
+    except Exception as e:
+        raise HTTPException(status_code=500, detail=str(e))
+
+@app.get("/indexes/{index_name}/documents", response_model=ListDocumentsResponse)
```

Comment:
can you give me an example about the "index_name" here? This variable may contain special characters, I am worried that it may not be able to be used as a URL.

Reply:
Updated /indexes endpoint to handle special characters. 
So now 


Raw Index Name | URL-Encoded Form (Sent by Client) | Decoded Form (on Server)
-- | -- | --
my_index | my_index | my_index
my index | my%20index | my index
index/name | index%2Fname | index/name
index@name | index%40name | index@name
index#1 | index%231 | index#1
index?query | index%3Fquery | index?query

Reply:
Can you move above comment to code comment?

Reply:
done